Assorted minor code improvements

superseeker13 opened the pull request:
Creating more specific regex in order to reject more strings without checking candidates.

superseeker13 committed:
Update massage regex to be move specific.

Creating more specific regex in order to reject more strings without checking candidates.

superseeker13 committed:
Update style.css

Remove style:.  Not parsable by all browsers.

superseeker13 committed:
Update ModuleSerializationProxy.java

Remove unused imports

superseeker13 committed:
Update LabelNode1.java

Remove unused imports

superseeker13 committed:
Update BoxesRunTime.java

Condense equals method

dwijnand commented:
> in order to reject more strings without checking candidates.

Why is this important?

superseeker13 commented:
It is not awfully important but having the regex there isn't either.  I figure that if there is a regex check at all, which can be time consuming,  it might as well be specific.Sent from my Galaxy
-------- Original message --------From: Dale Wijnand ***@***.***> Date: 5/17/21  4:51 PM  (GMT-05:00) To: scala/scala ***@***.***> Cc: superseeker13 ***@***.***>, Author ***@***.***> Subject: Re: [scala/scala] Update massage regex to be move specific. (#9637) 

in order to reject more strings without checking candidates.

Why is this important?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

som-snytt commented on src/repl-frontend/scala/tools/nsc/interpreter/shell/JavapClass.scala:
```diff
@@ -331,7 +331,7 @@ object Javap {
     require(arg startsWith "-")
     // arg matches opt "-foo/-f" if prefix of -foo or exactly -f
-    val r = """(-[^/]*)(?:/(-.))?""".r
+    val r = """(-[[^-/]|[a-z]]*)(?:/(-[a-z]))?""".r
```
this is just `helps.map(_.split("/"))`, except it is easier to cope with second element missing. If this is a good idea, might as well require `-[a-z]` on both sides. The alternation is especially useless because `[a-z]` is subsumed by `[^-/]` which means not a hyphen or slash.

superseeker13 replied:
Making the change now.

som-snytt commented on src/compiler/scala/tools/nsc/backend/jvm/LabelNode1.java:
```diff
@@ -13,7 +13,6 @@
 package scala.tools.nsc.backend.jvm;
 
 import scala.tools.asm.Label;
-import scala.tools.asm.tree.ClassNode;
```
What is this java? and have they no tooling to detect unused imports?

superseeker13 replied:
Not sure if they have no tooling or not but I've found more unused imports since.

som-snytt commented on src/library/scala/runtime/BoxesRunTime.java:
```diff
@@ -115,8 +115,7 @@ public static double unboxToDouble(Object d) {
     public static boolean equals(Object x, Object y) {
-        if (x == y) return true;
-        return equals2(x, y);
+        return (x == y) || equals2(x, y);
```
The context has excessive "early returns". I don't know whether this is a form of premature java optimization, or just to make java look bad. Here, I would count it as "ambient style" and either fix the entire file (having shown that it doesn't change the bytecode) or just leave it.

For example, `unboxToBoolean` could be rewritten `return b != null && something.booleanValue`. If the rewrite doesn't make it faster, does it make it easier to read?

superseeker13 replied:
I think that it does make it easier to read.

som-snytt reviewed (changes requested):
Since the helps are just literals, the massaged massage doesn't add much. But removing misleading unused leading imports is always an improvement.

superseeker13 committed:
Update JavapClass.scala

superseeker13 committed:
Update BoxesRunTime.java

Removing useless plus signs before method calls that return a numerical value and slightly speed up unboxToBoolean by simplifying a ternary to a shorted and.

superseeker13 commented:
Will fix in the morning.  I forgot to replace the positive sign with and absolute value call.  On secomd though that might be slower anyway.

superseeker13 commented:
Will check into optimizations and the StrictMath implementation tomorrow.

superseeker13 committed:
Update BoxesRunTime.java

SethTisue commented:
Is there test coverage? I'd be a bit reluctant to mess with the regex unless there's coverage to ensure we don't get it wrong.

som-snytt commented:
The regex change is trivial and IMHO neither adds or subtracts; at worse, it breaks REPL args which no one uses, I'm told.

I'd be more wary of fiddling with runtime without a lot of scrutiny and motivation. Boolean expressions are harder than regex because it's only 50/50 chance of getting it right.

Is there a reason intermediate values are pulled out into variables named `val`? Besides, "Oh, it's Java, I can use `val` here." I would not assume "oh that's dumb, there is a shorter and cleverer expression." Of course, sometimes there is no reason.

SethTisue commented:
There are now 8 commits in this PR:

* 59a0b8502036caf43747b6c26c8360b9890cde9c - purpose or value of this is unclear to me (and there is no commit message that explains it)
* 1eae0b1acf2cb04fc91535eaa7904395e0b9319f - minor code improvement, looks mergeable to me despite Som's trepidation
* 51d5267ed12833ed4629c86fc99103bc78f8c6d1, ef4509152e3091170c7b167a62570282bf7dfd8f - what happened here? do these cancel each other out?
* 38df9adb87ea3251afe46c9310e9bc93bcced170 - mergeable (minor code improvement)
* 369d504c50df5d37a20e0edc604846f376d4d2f2 - mergeable (unused import)
* 7bdc6a462b3e7a28dd97e0a63b31369220595224 - mergeable (unused import)
* de6029889615237acd68493c40dab0346fecb6bc - perhaps mergeable? are we sure there is no downside to the change? also, this is completely unrelated to the other commits and ought to be a separate PR

I've marked a number of commits as "mergeable", but we'd prefer they be squashed into fewer commits, for example the unused-imports commits don't need to be separate.

@superseeker13 we're fairly rigorous in this repo about keeping history clean — so unrelated changes go in separate PRs, and closely related changes should be together in a single commit

superseeker13 commented:
Understood, I will pull out unrelated change.

superseeker13 committed:
Merge branch 'scala:2.13.x' into 2.13.x

superseeker13 commented:
Closing pull request and planning on opening separate ones for the individual topics.
